### DISCUSSION TITLE
Issues with signup

### DISCUSSION THREAD

**Discussion by BPilot253** [pocketbase/pocketbase#1239] [Q&A] [answered]
I'm running Pocketbase behind a proxy on ExpressJS. Every time I try to create the first admin account, it timeouts after some time with the error "Failed to load the submitted data due to invalid formatting." and after enabling debugging mode, this shows in the terminal:
```
code=400, message=read tcp 127.0.0.1:8090->127.0.0.1:58668: i/o timeout, internal=read tcp 127.0.0.1:8090->127.0.0.1:58668: i/o timeout
```
What could the issue be?

**Comment by ganigeorgiev** [comment]
The "invalid formatting" error indicates most likely an issue with the body format (json or multipart/formdata). Could you provide a code sample how are you sending the request?

**Reply by BPilot253** [reply]
I'm interacting through the admin UI.

**Reply by ganigeorgiev** [reply]
In this case must be something with the proxy. Could you share your code?

**Reply by BPilot253** [reply]
This is the only code responsible for proxying.
```
const { createProxyMiddleware } = require("http-proxy-middleware");

router.use(createProxyMiddleware({
    target: "http://localhost:8090",
    changeOrigin: false
}));
```

**Reply by ganigeorgiev** [reply]
I've never used the express proxy module and I'm not sure what could be the reason for the error.

It could be related to the `localhost` target address if you are using node17+, try changing it to `http://127.0.0.1:8090` (see https://github.com/pocketbase/js-sdk/issues/21).

**Reply by ganigeorgiev** [reply]
You can also check the request in the browser's Inspect tool to see if it is sending to the correct path since the default backend url that the embedded js sdk is using is relative `../`, aka. if your proxied service is located at `https://myproxy.com/pb`, the working admin address would be `https://myproxy.com/pb/_/`.

**Reply by BPilot253** [reply]
It was apparently an issue from the proxy middleware I used. I switched to express-http-proxy instead and it works now. Thanks for the help though. 🙂

**Comment by BPilot253** [comment, marked as answer]
The issue was from the [http-proxy-middleware](https://www.npmjs.com/package/http-proxy-middleware) I used. Switching to [express-http-proxy](https://www.npmjs.com/package/express-http-proxy) fixed the issue.
